src/host/srvinit.cpp

```diff
@@ -276,11 +277,14 @@ static bool s_IsOnDesktop()
 
     if (fRecomputeOwner)

```

Comment:
Just for my own enlightenment: when there are two processes . . .

```
A (root)
B
```

and one exits (`A`):

```
B
```

do we reassign `B` as the root?

```
B (root)
```

Reply:
Yeah from what I can tell this code is at least supposed to accomplish this. If it doesn't do this, I believe this to be a bug, because there should always be a root process. I suppose the question is, can `LocateConsoleWindow()` ever return `nullptr`? I don't think it can except right before `ProcessExit`ing. In fact if it didn't return something here and we didn't have a root process, that'd be another bug. In short, this code isn't robust at all. Should I just call `HandleRootClientRemoved` at all times?